src/core/client/install/steps/containers/PermittedDomainsContainer.tsx

```diff
@@ -0,0 +1,79 @@
+import { FORM_ERROR } from "final-form";
+import React, { Component } from "react";
+import { FormData } from "../../containers/AppContainer";
+import PermittedDomains, {
+  PermittedDomainsForm,
+} from "../components/PermittedDomains";
+
+import { InstallInput } from "talk-framework/rest";
+import { InstallMutation, withInstallMutation } from "../mutations";
+
+interface PermittedDomainsContainerProps {
+  install: InstallMutation;
+  onGoToNextStep?: () => void;
+  onGoToPreviousStep?: () => void;
+  data: FormData;
+  onSaveData: (newData: {}) => void;
+}
+
+function shapeFinalData(data: FormData): InstallInput {
+  const {
+    organizationName,
+    organizationContactEmail,
+    organizationURL,
+    domains,
+    username,
+    password,
+    email,
+  } = data;
+
+  return {
+    tenant: {
+      organizationName,
+      organizationContactEmail,
+      organizationURL,
+      domains,
+    },
+    user: {
+      username,
+      password,
+      email,
+    },
+  };
+}
+
+class CreateYourAccountContainer extends Component<
+  PermittedDomainsContainerProps
+> {
+  private onGoToNextStep = () => {
+    if (this.props.onGoToNextStep) {
+      this.props.onGoToNextStep();
+    }
+  };
+  private onGoToPreviousStep = () => {
+    if (this.props.onGoToPreviousStep) {
+      this.props.onGoToPreviousStep();
+    }
+  };
+  private onSubmit: PermittedDomainsForm["onSubmit"] = async (input, form) => {
+    try {
+      const domains = input.domains.split(",");
+      this.props.onSaveData({ domains });
+      this.props.install(shapeFinalData(this.props.data));
```

Comment:
Because the data is being set in the state, and the `onSaveData` simply calls `setState` on the outer container, `this.props.data` does not actually have the data including the `domains`, so when you observe the last button on finishing the form, the `domains` field is never set.

Reply:
This is a race condition. I added an await since `setState` is not syncronous. Please check. Thanks!